### ISSUE TITLE
[Question]  PyTorch-compatible forward kinematics outputs?

### ISSUE THREAD

**Issue by umang-sarcomere** [isaac-sim/IsaacLab#2426]
I have a problem where I'm using the output of a neural network as a set of joint parameters, and the loss function I am using to train it depends on the robot link positions. As such, I need to be able to get gradients through the forward kinematics function. I've been using a separate differentiable forward kinematics library to do this so far, but it is seemingly much slower in training than previous projects I have done with Isaac Lab; ideally, I would like to transition my codebase to Lab.

So my question is: is there a way to get differentiable forward kinematics in Isaac Sim? If so, how would I achieve this? I know that the frame jacobians for an Articulation can be retrieved through `Articulation.root_physx_view.get_jacobians()`, but it doesn't seem to play well with a backwards pass in torch.

**Comment by RandomOakForest** [comment]
Thank you for posting this. I'll inquire with the team and post back here. Good question.

**Comment by umang-sarcomere** [comment]
Thank you! That would be greatly appreciated.

**Comment by RandomOakForest** [comment]
Following up on this. Unfortunately, I have confirmed this is not currently possible in Isaac Lab 2.1.0 or earlier. The team is aware of this, and we are constantly working to fill in this features. Thank you for your interest in Isaac Lab.

**Comment by mmacklin** [comment]
Hi there, we do have differentiable forward kinematics available in Warp, this can be connected to Torch through a custom autograd.op for optimization as in this inverse kinematics example:

https://github.com/NVIDIA/warp/blob/main/warp/examples/optim/example_inverse_kinematics_torch.py

(see `eval_fk()` method).

Warp is available in Isaac Lab, although it is not integrated with the Articulation class directly.

**Comment by umang-sarcomere** [comment]
@RandomOakForest thank you for the update! I'll keep an eye on future Isaac Lab versions.

@mmacklin The methodology in the Warp example looks like it would work well for my use, thank you for pointing it out! I am not super familiar with using Warp directly, but I'll give it a shot and see how it goes.
